src/profiles_management/helpers/k8s.py

```diff
@@ -0,0 +1,48 @@
+"""Generic helpers for manipulating K8s objects, via lightkube."""
+
+from lightkube import Client
+from lightkube.generic_resource import GenericGlobalResource, GenericNamespacedResource
+
+client = Client(field_manager="profiles-automator-lightkube")
+
+
+def get_name(res: GenericNamespacedResource | GenericGlobalResource) -> str:
+    """Return the name from generic lightkube resource.
+
+    Args:
+        res: The resource to get it's name from metadata.name
+
+    Raises:
+        ValueError: if the object doesn't have metadata or metadata.name
+
+    Returns:
+        The name of the object from its metadata.
+    """
+    if not res.metadata:
+        raise ValueError("Coldn't detect name, object has no metadata: %s" % res)
+
+    if not res.metadata.name:
+        raise ValueError("Couldn't detect name, object has no name field: %s" % res)
```

Comment:
In which scenarios will a resource not have metadata? Also, perhaps [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) is not the best exception to raise here. Maybe you'd want to consider a custom exception or rely on Lightkube exceptions for it.

Reply:
None AFAIK, but `lightkube`'s typings for some reason the `.metadata` field to be `ObjectMeta | None`.

So the linter would complain if I'd directly try to do `res.metadata.name`

Regarding the exception type, unfortunately lightkube [doesn't have a good one](https://github.com/gtsystem/lightkube/blob/master/lightkube/core/exceptions.py). If we'd create a custom exception, then maybe it would be something like `InvalidObjectMeta`? What do you suggest?